addons/sale_project/models/sale_order.py

```diff
@@ -36,10 +36,12 @@ def _compute_visible_project(self):
     @api.depends('order_line.product_id', 'order_line.project_id')
     def _compute_project_ids(self):
         is_project_manager = self.user_has_groups('project.group_project_manager')
+        Project = self.env['project.project']
```

Comment:
Sorry I didn't pay enough attention during my first review. In fact we should avoid doing a search in a for loop because we will have 1 additional query per Sales Order contained in `self`.
To avoid that, I suggest to search all projects for all Sales Order contained in `self` and then create a dict to get the projects per SO.
```suggestion
        projects = self.env['project.project'].search([('sale_order_id', 'in', self.ids)])
        projects_per_so = defaultdict(lambda: self.env['project.project'])
        for project in self:
            projects_per_so[project.sale_order_id.id] |= project
```